forge-gui/res/cardsfolder/upcoming/the_day_of_the_doctor.txt

```diff
@@ -0,0 +1,17 @@
+Name:The Day of the Doctor
+ManaCost:3 R W
+Types:Enchantment Saga
+K:Saga:4:DBDig,DBDig,DBDig,DBChoose
+SVar:DBDig:DB$ DigUntil | Defined$ You | Valid$ Card.Legendary | FoundDestination$ Exile | RevealedDestination$ Exile | RememberRevealed$ True | ImprintFound$ True | SubAbility$ DBEffect | SpellDescription$ Exile cards from the top of your library until you exile a legendary card. You may play that card for as long as CARDNAME remains on the battlefield. Put the rest of those exiled cards on the bottom of your library in a random order.
+SVar:DBEffect:DB$ Effect | StaticAbilities$ MayPlay | RememberObjects$ Imprinted | Duration$ UntilHostLeavesPlay | SubAbility$ DBRestRandomOrder
+SVar:MayPlay:Mode$ Continuous | Affected$ Card.IsRemembered+nonLand | MayPlay$ True | EffectZone$ Command | AffectedZone$ Exile | Description$  You may play that card for as long a CARDNAME remains on the battlefield.
+SVar:DBRestRandomOrder:DB$ ChangeZoneAll | ChangeType$ Card.IsRemembered+IsNotImprinted | Origin$ Exile | Destination$ Library | LibraryPosition$ -1 | RandomOrder$ True | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True | ClearImprinted$ True
+SVar:DBChoose:DB$ ChooseCard | Defined$ You | MinAmount$ 0 | Amount$ 3 | Choices$ Doctor | ChoiceTitle$ Choose up to three doctors | SubAbility$ DBGenericChoice
+SVar:DBGenericChoice:DB$ GenericChoice | Choices$ DBExileAll,DBCleanupBis
+SVar:DBExileAll:DB$ ChangeZoneAll | ChangeType$ Creature.nonChosenCard | Origin$ Battlefield | Destination$ Exile | SubAbility$ DBDamage | SpellDescription$ Exile all other creatures. If you do, The Day of the Doctor deals 13 damage to you.
```

Comment:
Ugh, this is pretty gross. This generic choice might be the best way to do it. I can think of two other possible ways that might be less lines.
1 –
ChooseCard
ChangeZoneAll nonChosen + Optional + RememberChanged
DealDamage + Condition (Remembered)
 - Not sure what happens in case there are no "other creatures" to exile??
 
 2 –
 Choosecard
 DealDamage + UnlessCost Exile all nonchosen creatures + UnlessSwitched etc
  - Not quite sure the engine can do this yet?

@tool4ever any thoughts on this? It's definitely a cost but we handle these in various ways with this kind of templating.

Reply:
To clarify the ability is gross – not your implementation 😄